### PR TITLE
Missions: Fix missions, install script

### PR DESCRIPTION
**Pull request by jpanikulam**
- Make install/update script much less shity

### CODE REVIEW

File: command/sub8_missions/readme.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Sub Mission Manager
 # Running the sub
 
 ```shell
+    # Only works if you're the sub!
     roslaunch sub8_launch sub8.launch
 ```
```
Comment (tessbianchi) [inline command/sub8_missions/readme.md:13]: Add use the sim if you aren't using the sub

File: scripts/get_dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            instlog "$folder not a git repository"
+            continue;
+        fi
+
+        LOCAL_BRANCH=`git name-rev --name-only HEAD`
+        TRACKING_BRANCH=`git config branch.$LOCAL_BRANCH.merge`
+        TRACKING_REMOTE=`git config branch.$LOCAL_BRANCH.remote`
+        REMOTE_URL=`git config remote.$TRACKING_REMOTE.url`
+        if python -c "import re; _, have_url = re.split('https://github.com|git@github.com:', '$REMOTE_URL'); _, want_url = re.split('https://github.com|git@github.com:', '$INSTALL_URL'); exit(have_url == want_url)"; then
+            instlog "Already have package at url $INSTALL_URL"
```
Comment (tessbianchi) [inline scripts/get_dependencies.sh:50]: You dirty dirty man

**Comment by tessbianchi** [comment]
Approved

File: command/sub8_missions/sub8/tx_sub.py [hunk 3]
```diff
@@ -33,14 +33,15 @@ class _Sub(object):
...
+    @util.cancellableInlineCallbacks
     def _init(self):
-        self._moveto_action_client = action.ActionClient(self._node_handle, 'moveto', MoveToAction)
-        self._odom_sub = self._node_handle.subscribe('odom', Odometry)
-        self._trajectory_sub = self._node_handle.subscribe('trajectory', PoseTwistStamped)
-        self._trajectory_pub = self._node_handle.advertise('trajectory', PoseTwistStamped)
+        self._moveto_action_client = yield action.ActionClient(self._node_handle, 'moveto', MoveToAction)
+        self._odom_sub = yield self._node_handle.subscribe('odom', Odometry)
+        self._trajectory_sub = yield self._node_handle.subscribe('trajectory', PoseTwistStamped)
+        self._trajectory_pub = yield self._node_handle.advertise('trajectory', PoseTwistStamped)
```
Comment (forrestv) [inline command/sub8_missions/sub8/tx_sub.py:41]: FYI, yielding on these isn't required. It doesn't hurt, of course. It's kind of annoying that there's not some good way to know/indicate what returns a Deferred and what doesn't. You could postfix functions/methods that return a Deferred with "_df" or something, but that's ugly.

**Pull request merged**